assembl/static2/js/app/components/login/signupForm.jsx

```diff
@@ -96,6 +140,25 @@ class SignupForm extends React.Component {
                 onChange={this.handleInput}
               />
             </FormGroup>
+
+            {hasTermsAndConditions && (
+              <FormGroup>
+                <Checkbox checked={this.state.checked} type="checkbox" onChange={this.toggleCheck} required inline>
+                  <Translate value="termsAndConditions.iAccept" />
+                  <a
+                    onClick={(e) => {
+                      e.preventDefault();
+                      const Terms = (
+                        <TermsForm handleAcceptButton={this.handleAcceptButton} isChecked={this.state.checked} lang={lang} />
+                      );
+                      displayCustomModal(Terms);
+                    }}
+                  >
+                    <Translate value="termsAndConditions.link" className="terms-link" />
+                  </a>
```

Comment:
You should put the `<a` in a `termsLink` variable and instead of
`<Translate value="termsAndConditions.iAccept" />`
use
`{ I18n.t('termsAndConditions.link', { termsLink: termsLink }) }` and for the translation
`iAccept: "I have read and I accept the {termsLink}"`
This will be much better for internationalization. In some languages, the link will be before the rest of the text.

Reply:
To be tested, may not work at all.